### DISCUSSION TITLE
v0.3.0-three-phase

### DISCUSSION THREAD

**Discussion by David00** [David00/rpi-power-monitor#94] [Announcements]
This discussion thread is intended for the three-phase variant of the project.  I sent an email out to everyone on my list that was interested in the coming 3-phase update, and pointed them here for further discussion and feedback.

If you did not receive the email, I must not have had you on my list, but that's ok! You can get involved right here in this thread and test the latest 3-phase release by reviewing the setup instructions on the [3-Phase Setup Guide](https://david00.github.io/rpi-power-monitor/docs/v0.3.0-beta/3-phase) in the docs.

**Comment by cocanut** [comment]
Dear David00 and community,
thank you very much for your efforts and the super documentation! I live in germany where - to my knowledge - almost every household has three different phases, each at about 50 Hz and 230 V. As @David00 described in the [documentation](https://david00.github.io/rpi-power-monitor/docs/v0.3.0-beta/3-phase) the voltage of the three phases (lets call them A, B and C) is usually not identical. Does anyone know why the voltage differs on each phase and how big this error is?

I would like to try out the power monitor to optimize the energy consumption of a culture-center (cinema, café, live music, all types of events) on the countryside of germany. So I think we can "afford" some error, as we are not so much interested in super accuracy about the absolut energy consumption of each sensor but more interested in the differences between the sensors. I.e. with the power monitor I can compare the total energy consumption of the "ventilation system" which sits on phase A, B and C and the "light and sound machinery of the concert hall" which justs sits on phase A and B.

**Reply by David00** [reply]
Hi @cocanut, sorry for the delay in getting back to you. The reason the voltage differs between the three phases has many possibilities and can be quite complex, but one key factor is the imbalanced loading of each of the three phases.  The imbalance isn't just referring to your own house's panel - but all of your neighbors that share the same main feed as well. If a bunch of houses in your local distribution feed have large loads on phase A, they're going to bring the voltage down for your phase A, and so on.  I don't know if this same logic applies to upstream of the substation or not, but I wouldn't be surprised if it did.

I had some assistance from a gentleman from Australia in testing and measuring the accuracy of my 3-phase variant, and he took 20 manual voltage measurements of each of his three phases across two days to gauge the level of error between the phases.  The largest voltage spread we saw was 7.8V between the highest and lowest phase (that is, one phase read 223.3, the other read 226.5, and the last phase read 231.1. The average voltage spread between all three phases was only 2.6V.

From the perspective of this project, with its single voltage input source, the best possible accuracy would have been provided by putting the voltage transformer on the phase measuring 226.5V, because it was in the middle of the low and high readings.  However, I don't have any data to suggest (or refute) that picking this phase to collect voltage measurements from would have remained the best phase.  

However, the voltage can change dramatically on each phase. In all of his measurements , phase 1 varied by 26.7V, phase 2 varied by 20V, and phase 3 varied by 23.4V.  This isn't too much of a problem, but it does demonstrate that we can't just hardcode a voltage level for a phase into the software.


> I would like to try out the power monitor to optimize the energy consumption of a culture-center (cinema, café, live music, all types of events) on the countryside of germany. So I think we can "afford" some error, as we are not so much interested in super accuracy about the absolut energy consumption of each sensor but more interested in the differences between the sensors. I.e. with the power monitor I can compare the total energy consumption of the "ventilation system" which sits on phase A, B and C and the "light and sound machinery of the concert hall" which justs sits on phase A and B.

This is a perfect use case for the 3-phase variant. You'll still be able to see the accurate amperage readings from each of the sensors, and you can use these readings to balance the load on your phases.  The voltage variance only impacts the real/apparent/reactive power (Watts/Volt-Amperes/Volt-Amperes-Reactive) and energy (Wh) calculations.
